warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExpandMultiNormalizedTerms.java

```diff
@@ -92,6 +92,9 @@ public ExpandMultiNormalizedTerms(ShardQueryConfiguration config, MetadataHelper
             throw new DatawaveFatalQueryException(qe);
         }
         
+        // inject reference nodes when needed
+        script = (T) EnsureReferenceNodesVisitor.ensureReferences(script);
```

Comment:
Why do we need this here?  Is a previous visitor breaking this contract?

Reply:
ExpandMultiNormalizedTerms can produce a reference expression without a reference node

Reply:
But you call it before the visitor is actually called?